Zotero/Scenes/Detail/CopyBibliography/Views/CopyBibliographyViewController.swift

```diff
@@ -0,0 +1,171 @@
+
+        let webView = WKWebView()
+        webView.isHidden = true
+        view.insertSubview(webView, at: 0)
+        self.webView = webView
+
+        NSLayoutConstraint.activate([
+            overlayBody.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            overlayBody.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            overlayBody.leadingAnchor.constraint(greaterThanOrEqualTo: view.leadingAnchor, constant: 60),
+            view.trailingAnchor.constraint(greaterThanOrEqualTo: overlayBody.trailingAnchor, constant: 60),
+            stackView.topAnchor.constraint(equalTo: overlayBody.topAnchor, constant: 20),
+            stackView.leadingAnchor.constraint(equalTo: overlayBody.leadingAnchor, constant: 20),
+            overlayBody.bottomAnchor.constraint(equalTo: stackView.bottomAnchor, constant: 20),
+            overlayBody.trailingAnchor.constraint(equalTo: stackView.trailingAnchor, constant: 20)
+        ])
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = UIColor(white: 0, alpha: 0.1)
+        view.alpha = 0
+        setupObserving()
+        viewModel.process(action: .preload(webView))
+
+        func setupObserving() {
+            viewModel.stateObservable
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { state in
+                    update(state: state)
+                })
+                .disposed(by: disposeBag)
+
+            func update(state: CopyBibliographyState) {
+                if state.processingBibliography {
+                    showOverlay(state: .processing)
+                } else if let error = state.error {
+                    if let error = error as? CitationController.Error, error == .styleOrLocaleMissing {
+                        coordinatorDelegate?.showMissingStyleError(using: nil)
+                        hideOverlay()
+                    } else {
+                        showOverlay(state: .error(L10n.Errors.Items.generatingBib))
+                        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(1500)) { [weak self] in
+                            self?.hideOverlay()
+                        }
+                    }
+                } else {
+                    hideOverlay()
+                }
+            }
+        }
+    }
+
+    // MARK: - Actions
+    private func showOverlay(state: OverlayState) {
+        switch state {
+        case .processing:
+            overlayText.text = L10n.Items.generatingBib
+            overlayActivityIndicator.isHidden = false
+            overlayActivityIndicator.startAnimating()
+            overlayErrorIcon.isHidden = true
+
+        case .error(let message):
+            overlayText.text = message
+            overlayActivityIndicator.stopAnimating()
+            overlayErrorIcon.isHidden = false
+        }
+
+        view.layoutIfNeeded()
+
+        UIView.animate(withDuration: 0.2) {
```

Comment:
Maybe add a guard in case the overlay is already visible.

Reply:
A guard against `view.alpha` has been added.
Check https://github.com/zotero/zotero-ios/pull/827/commits/7e5eccebcb0a2044d80477c3a59d9ada02329699